packages/xod-client/src/editor/reducer.js

```diff
@@ -468,6 +468,25 @@ const editorReducer = (state = {}, action) => {
         action.payload.tabId,
         state
       );
+    case EAT.EDITOR_OPEN_IMPLEMENTATION_CODE:
+      return R.over(
+        currentTabLens,
+        R.merge(R.__, {
+          type: TAB_TYPES.IMPLEMENTATION_CODE,
+          isReadOnly: action.payload.isReadOnly,
+        }),
+        state
+      );
+    case EAT.EDITOR_CLOSE_IMPLEMENTATION_CODE:
+      return R.over(
+        currentTabLens,
+        R.compose(
+          // TODO: what if we are in debug mode?
```

Comment:
Ideally, we should be switched back to the tab/place/area where we initiated the code editor opening.

Not a big deal though.